### PR TITLE
Update to react-scripts 5

### PR DESCRIPTION
**Pull request by RDIL**
Successor to #424 (easier to reopen with changes then deal with merge conflicts)

- Upgrade to react-scripts 5
- This time with a little bit less patching action, but still a little 😞

### CODE REVIEW

File: .yarn/patches/react-scripts-npm-5.0.1-d06bd2d5ad.patch [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+           babelRuntimeRegenerator,
+         ]),
+       ],
++      fallback: {
++        assert: require.resolve("assert-browserify"),
++        buffer: require.resolve("buffer-browserify"),
++        crypto: require.resolve("crypto-browserify"),
++        stream: require.resolve("stream-browserify"),
++      },
+     },
```
Comment (RDIL) [inline .yarn/patches/react-scripts-npm-5.0.1-d06bd2d5ad.patch:15]: Some dependencies, such as sjcl, use Node built-in libraries, which isn't possible in the browser.

webpack 5 *no longer provides automatic polyfills*, so this is required in order to use those dependencies.

File: .yarnrc.yml [hunk 2]
```diff
@@ -1,7 +1,23 @@
...
+# Some dependencies used by this project use Node built-ins, which aren't
+# available on the web. `react-scripts@5` uses webpack 5, which no longer
+# provides polyfills, so we need to manually inject the polyfills in ourselves.
+packageExtensions:
+  react-scripts@*:
+    dependencies:
+      assert-browserify: '*'
+      buffer-browserify: '*'
+      crypto-browserify: '*'
+      stream-browserify: '*'
```
Comment (RDIL) [inline .yarnrc.yml:12]: Ensures the polyfills are installed.

File: package.json [hunk 3]
```diff
@@ -47,7 +47,7 @@
     "react-stripe-elements": "^6.1.2",
     "react-vis": "^1.11.7",
     "recharts": "^1.8.5",
-    "recoil": "^0.5.2",
+    "recoil": "^0.7.4",
```
Comment (RDIL) [inline package.json:50]: Required for the `@types/react` upgrade.

**Review by RDIL — COMMENTED** [review]
Some extra context for reviewers.

File: src/scenes/__tests__/__snapshots__/NotFound.test.tsx.snap [hunk 4]
```diff
@@ -24,7 +24,7 @@ exports[`renders correctly 1`] = `
       Page not found! 
     </h2>
     <a
-      className="MuiTypography-root MuiTypography-h3 MuiLink-root MuiLink-underlineAlways css-77jlxd-MuiTypography-root-MuiLink-root"
+      className="MuiTypography-root MuiTypography-h3 MuiLink-root MuiLink-underlineAlways css-1mgvoth-MuiTypography-root-MuiLink-root"
```
Comment (RDIL) [inline src/scenes/__tests__/__snapshots__/NotFound.test.tsx.snap:27]: For some reason, this upgrade changed this snapshot.  This makes the tests pass.

**Comment by RDIL** [comment]
@fkorotkov review ping :)

**Comment by fkorotkov** [comment]
Sorry we don't have time to review at the moment. This change is not bringing new functionality/improvements but requires a lot of manual testing since it affect the whole app. Once we have time to invest in improving the UI we'll test this change as well.

**Review by bitcoinfinancier — APPROVED** [review]

**Pull request closed without merging**